src/pytorch_lightning/core/module.py

```diff
@@ -291,12 +291,12 @@ def truncated_bptt_steps(self, truncated_bptt_steps: int) -> None:
         self._truncated_bptt_steps = truncated_bptt_steps
 
     @property
-    def logger(self) -> Optional[Logger]:
+    def logger(self) -> Optional[Union[Logger, TensorBoardLogger]]:
```

Comment:
Did you make this change because of mypy? This shouldn't be necessary, no?

Reply:
Logger here is the Logger base class from PyTorch Lightning
TensorBoardLogger here is subclassing FabricTensorBoardLogger which is subclassing the base class in Fabric

So these don't have the same base class, and therefore need to be included for typing separately. This is a consequence of your request to reuse code through the inheritance.

Reply:
This is passing mypy locally for me: [`2e90ddf` (#16121)](https://github.com/Lightning-AI/lightning/pull/16121/commits/2e90ddf7d0738bb9f24c268e93a5330dd3ff4c2f)

It just changes the base import to Fabric's. Do you see any issues with this?